docker/dockerfiles/Dockerfile.sam2

```diff
@@ -0,0 +1,61 @@
+FROM nvcr.io/nvidia/cuda:12.4.1-cudnn-devel-ubuntu22.04
+
+WORKDIR /app
+
+RUN rm -rf /var/lib/apt/lists/* && apt-get clean && apt-get update -y && DEBIAN_FRONTEND=noninteractive apt-get install -y \
+    ffmpeg \
+    libxext6 \
+    libopencv-dev \
+    uvicorn \
+    python3-pip \
+    git \
+    libgdal-dev \
+    wget \
+    && rm -rf /var/lib/apt/lists/*
+
+COPY requirements/requirements.sam2.txt \
+    requirements/requirements.http.txt \
+    requirements/_requirements.txt \
+    requirements/requirements.gpu.txt \
+    requirements/requirements.sdk.http.txt \
+    ./
+
+RUN python3 -m pip install --upgrade pip
+RUN python3 -m pip install --extra-index-url  https://download.pytorch.org/whl/cu141  \
+    -r _requirements.txt \
+    -r requirements.sam2.txt \
+    -r requirements.http.txt \
+    -r requirements.gpu.txt \
+    -r requirements.sdk.http.txt \
+    --upgrade \
+    && rm -rf ~/.cache/pip
+
+# Install setup.py requirements for flash_attn
+RUN python3 -m pip install packaging==24.1  && rm -rf ~/.cache/pip
+
+# Install flash_attn required for Paligemma and Florence2
+# RUN python3 -m pip install -r requirements.pali.flash_attn.txt --no-build-isolation && rm -rf ~/.cache/pip
+WORKDIR /sam
+RUN apt-get install -y gcc
```

Comment:
I'd add gcc to the list of packages in first `RUN` command (unless there is good reason to install it separately here)